connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
I'm not sure I can call it a bug. Forced HE was planned to solve the WiFi + cellular use case as it's the most common and was never considered for Blouetooth/USB. Bluetooth and USB use cases are I guess not that common, but you are right that they could be solve similarly. Feel free to create JIRA issues for fixing that, or just fix it.  
Concerning this, let's leave it as it is now as you are right it will cover all cases. I only have the comment about ConnectivityManager.